third_party/gfxreconstruct/framework/decode/vulkan_dive_consumer_base.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "decode/vulkan_dive_consumer_base.h"
+#include "decode/custom_vulkan_struct_to_json.h"
+
+#include "generated/generated_vulkan_enum_to_json.h"
+
+#include "util/json_util.h"
+#include "util/platform.h"
+#include "util/file_path.h"
+#include "util/to_string.h"
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(decode)
+
+using namespace util::platform;
+
+VulkanExportDiveConsumerBase::VulkanExportDiveConsumerBase() {}
```

Comment:
is there a specific reason you want to keep the empty constructor? or maybe we could use VulkanExportDiveConsumerBase() = default;  in the header?